src/main/scala/com/gu/zuora/creditor/CreditTransferService.scala

```diff
@@ -47,6 +36,27 @@ class CreditTransferService(command: CreateCreditBalanceAdjustmentCommand)(impli
         s"${reportLine.subscriptionName} as its balance is not negative or some other piece of information is missing")
     }
   }
+}
+
+class CreditTransferService(
+                             adjustCreditBalance: Seq[CreateCreditBalanceAdjustment] => ZuoraCreditBalanceAdjustmentRes,
+                             downloadGeneratedExportFile: ExportId => Option[RawCSVText],
```

Comment:
I know its common practice in alot of these projects but passing in functions like this make the code difficult to follow, using a trait allows you to find implementations more easily.

Reply:
Fair point.
I will leave ti for now as is because changing that will require some changes in tests as well

Initially it was a case class with implicit zuora restClient
I wen to to passing functions as then you can pass stub functions without a need of mocking rest client